### PR TITLE
primed nukes explode if blobbed

### PR DESCRIPTION
**Pull request by TDSSS**
## What Does This PR Do
This PR makes it so that when a blob eats an active, primed nuke that's counting down, it causes the nuke to explode.

## Why It's Good For The Game
Nukes are set to end the round, a blob eating the nuke means the round drags on and since midround blobs don't have a win condition it just forces admins to send in a deathsquad with another nuke while all the ghosts watch.

It also removes a source of confusion regarding rules on is it allowed for blobs to rush nukes or not, since it makes it into a pretty bad idea now.

## Changelog
:cl:
tweak: blobs eating active nukes makes them blow up early.
/:cl:

### CODE REVIEW

**Comment by Darkmight9** [comment]
oh yes thank you please

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 1]
```diff
@@ -347,6 +347,10 @@ GLOBAL_VAR(bomb_set)
 /obj/machinery/nuclearbomb/blob_act(obj/structure/blob/B)
 	if(timing == -1.0)
 		return
+	if(timing)	//boom
+		spawn
```
Comment (farie82) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: spawn(0) is usually used.

Comment (TDSSS) [reply code/game/gamemodes/nuclear/nuclearbomb.dm]: I copied it directly from the normal nuke explosion code. What's the practical difference between spawn and spawn(0)?

Comment (farie82) [reply code/game/gamemodes/nuclear/nuclearbomb.dm]: Should be the same I believe. Both are done after the proc is done "async". More a styling thing

Comment (Fox-McCloud) [reply code/game/gamemodes/nuclear/nuclearbomb.dm]: use invoke async. Spawns are extremely discourage these days.

Comment (TDSSS) [reply code/game/gamemodes/nuclear/nuclearbomb.dm]: Alright. Want me to change it in the process() proc too while I'm there?

**Comment by TDSSS** [comment]
Should use invoke async now

**Comment by daylight2** [comment]
Seems like this would make rushing the nuke even more of a priority, because if you wait too long, you're fucked

**Comment by Cazdon** [comment]
Well, all it takes now is for a blob mouse to pop inside the vault for the round to instantly end, I'm not sure this is the best option and will just lead to another set of problems regarding that.

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 2]
```diff
@@ -347,6 +346,9 @@ GLOBAL_VAR(bomb_set)
 /obj/machinery/nuclearbomb/blob_act(obj/structure/blob/B)
 	if(timing == -1.0)
 		return
+	if(timing)	//boom
+		INVOKE_ASYNC(src, .proc/explode)
```
Comment (Kyep) [inline code/game/gamemodes/nuclear/nuclearbomb.dm:350]: Will this still run if the object is qdel()ed?

Comment (TDSSS) [reply code/game/gamemodes/nuclear/nuclearbomb.dm:350]: It won't work correctly if the object gets qdeled. That's why I have it return early so the nuke isn't deleted.

Comment (Kyep) [reply code/game/gamemodes/nuclear/nuclearbomb.dm:350]: Yeah, fair.

**Pull request merged**